### PR TITLE
add wireless thermostat accessory Heizkörperthermostat Basic 6256/1-WL

### PR DESCRIPTION
**Pull request by dn-falk**
# Description
This pull request adds support for radio heating thermostats of type "Heizkörperthermostat Basic 6256/1-WL“. Possibly also for the „Heizkörperthermostat Comfort 6256/2-WL“, but unfortunately I do not have this one for testing. The wireless thermostats use different datapoints than the existing thermostat implementation, so I added a new accessory category, took the existing implementation and adjusted the datapoints.

# API sample
```
{
  "00000000-0000-0000-0000-000000000000": {
    "devices": {
      "XXXXXXXXXX": {
        "floor": "02",
        "room": "09",
        "interface": "RF",
        "displayName": "Thermostat Büro",
        "unresponsive": false,
        "channels": {
          "ch0000": {
            "displayName": "Thermostat Büro",
            "functionID": "3e",
            "inputs": {
              "idp0000": {
                "pairingID": 52,
                "value": "invalid"
              },
              "idp0001": {
                "pairingID": 56,
                "value": "0"
              },
              "idp0002": {
                "pairingID": 54,
                "value": "0"
              },
              "idp0003": {
                "pairingID": 57,
                "value": "invalid"
              },
              "idp0004": {
                "pairingID": 58,
                "value": "0"
              },
              "idp0005": {
                "pairingID": 66,
                "value": "0"
              },
              "idp0006": {
                "pairingID": 7,
                "value": "0"
              },
              "idp0007": {
                "pairingID": 4,
                "value": "0"
              },
              "idp0008": {
                "pairingID": 53,
                "value": "0"
              },
              "idp0009": {
                "pairingID": 320,
                "value": "21"
              },
              "idp000a": {
                "pairingID": 304,
                "value": "invalid"
              },
              "idp000b": {
                "pairingID": 332,
                "value": "0"
              }
            },
            "outputs": {
              "odp0000": {
                "pairingID": 51,
                "value": "7"
              },
              "odp0001": {
                "pairingID": 52,
                "value": "0"
              },
              "odp0002": {
                "pairingID": 56,
                "value": "0"
              },
              "odp0003": {
                "pairingID": 54,
                "value": "96"
              },
              "odp0004": {
                "pairingID": 57,
                "value": "invalid"
              },
              "odp0005": {
                "pairingID": 66,
                "value": "0"
              },
              "odp0006": {
                "pairingID": 58,
                "value": "0"
              },
              "odp0007": {
                "pairingID": 304,
                "value": "22.8"
              },
              "odp0008": {
                "pairingID": 331,
                "value": "0"
              },
              "odp0009": {
                "pairingID": 333,
                "value": "0"
              },
              "odp000a": {
                "pairingID": 305,
                "value": "0"
              },
              "odp000b": {
                "pairingID": 273,
                "value": "0"
              },
              "odp000c": {
                "pairingID": 59,
                "value": "21"
              },
              "odp000d": {
                "pairingID": 68,
                "value": "0"
              }
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

**Comment by dn-falk** [comment]
@superyaro Hi Yaro, if you want we can go through the pull requests together. Just get in touch. Greetings Dennis

File: src/accessories/BuschJaegerWirelessThermostatAccessory.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        })
+        .on('get', this.getTemperatureDisplayUnits.bind(this));
+
+    /*
+     * If we set TargetHeatingCoolingState to OFF the accessory will set the
+     * target heating temperature to 35°C or 7°C. We will store the last target temperature
+     * in this variable do prevent incorrect readouts. The real target temperature
+     * is re-set correctly when the TargetHeatingCoolingState is set back to AUTO.
+     */
+    this.lastTargetTemperature = 0;
```
Comment (superyaro) [inline src/accessories/BuschJaegerWirelessThermostatAccessory.js:49]: Great solution! it always annoys me... Can we adopt this to the wireless Thermostat as well?

**Comment by superyaro** [comment]
Dennis,
can you check the changelog please. I pushed a 3.0.5-beta.1 to NPM, if it's ok for you I release it. Please check it on your system since I don't have the actors running (yet),

thank you for the big contrib, it makes me feel less alone ;)

J

File: src/accessories/BuschJaegerWirelessThermostatAccessory.js (hunk 1 above)
Comment (superyaro) [reply src/accessories/BuschJaegerWirelessThermostatAccessory.js:49]: my fault: it's already in "BuschJaegerThermostatAccessory" but doesn't seem to work...

**Comment by superyaro** [comment]
@dennis-nikolas-falk We should extend BuschJaegerWirelessThermostatAccessory.js from BuschJaegerThermostatAccessory.js in the future, I would like to change the Accessories to a "class"/"extend" system like in  BuschJaegerSmokeSensorAccessory:
`class BuschJaegerSmokeSensorAccessory extends BuschJaegerAccessory`
what do you think about it?

Jaro

File: src/accessories/BuschJaegerWirelessThermostatAccessory.js (hunk 1 above)
Comment (dn-falk) [reply src/accessories/BuschJaegerWirelessThermostatAccessory.js:49]: It looked like this worked for my wireless thermostats. I don't have any normal thermostats to compare with. I will check this more closely in the near future. Maybe we can then rework that again.

**Comment by dn-falk** [comment]
> @dennis-nikolas-falk We should extend BuschJaegerWirelessThermostatAccessory.js from BuschJaegerThermostatAccessory.js in the future, I would like to change the Accessories to a "class"/"extend" system like in BuschJaegerSmokeSensorAccessory: `class BuschJaegerSmokeSensorAccessory extends BuschJaegerAccessory` what do you think about it?
> 
> Jaro

Yes I think that would be the better way. But unfortunately I don't have that much experience with JavaScript/TypeScript and Node.js/NPM. I work more with Java. But I would still like to support the project.

**Comment by dn-falk** [comment]
> Dennis, can you check the changelog please. I pushed a 3.0.5-beta.1 to NPM, if it's ok for you I release it. Please check it on your system since I don't have the actors running (yet),
> 
> thank you for the big contrib, it makes me feel less alone ;)
> 
> J

I will check the version and let you know. Thanks for the note in the changelog :)

**Comment by dn-falk** [comment]
@superyaro Looks good to me. Wireless thermostat, wireless window contact and binary inputs work. You can publish this version as stable/latest if you like. Suggestion: I think it would be a good idea to have an issue board here in the project.

**Comment by superyaro** [comment]
opened Issues

**Pull request merged**